src/components/discreteSlider/index.tsx

```diff
@@ -0,0 +1,94 @@
+import React from 'react';
+import {
+  Slider as ChakraSlider,
+  SliderMark as ChakraSliderMark,
+  SliderFilledTrack,
+  SliderThumb,
+  SliderTrack,
+} from '@chakra-ui/react';
+
+const emptyItemOffset = 0;
+const firstItemOffset = 1;
+
+type SliderMark<T> = {
+  stepValue?: number;
+  label: string;
+  value: T;
+};
+
+type SliderProps<T> = {
+  applyIndentation?: boolean;
+  marks: SliderMark<T>[];
+  value: T;
+  onValueChanged: (arg: T) => void;
+};
+
+const getSliderStepValue = <T,>(
+  marks: SliderMark<T>[],
+  applyIndentation: boolean,
+  value: T,
+) => {
+  const selectedMarkIndex = marks.findIndex((mark) => mark.value === value);
+  const stepIndexOffset = applyIndentation ? firstItemOffset : emptyItemOffset;
+  if (selectedMarkIndex < emptyItemOffset) {
+    return stepIndexOffset;
+  }
+  return selectedMarkIndex + stepIndexOffset;
+};
+
+const getSliderMarks = <T,>(
+  marks: SliderMark<T>[],
+  applyIndentation: boolean,
+) =>
+  marks.map((mark, index) => ({
+    ...mark,
+    stepValue: index + Number(applyIndentation),
+  }));
+
+const DiscreteSlider = <T,>({
+  marks,
+  applyIndentation = true,
+  onValueChanged,
+  value,
+}: SliderProps<T>) => {
+  const sliderStepValue = getSliderStepValue(marks, applyIndentation, value);
+
+  const sliderMarks = getSliderMarks(marks, applyIndentation);
+
+  const handleOnChange = (newStepValue: number) => {
+    const newSliderMark = sliderMarks.find(
+      (mark) => mark.stepValue === newStepValue,
+    )?.value;
+
+    onValueChanged(newSliderMark as NonNullable<T>);
+  };
+
+  return (
+    <ChakraSlider
+      step={1}
+      max={sliderMarks.length - Number(!applyIndentation)}
```

Comment:
There is an opportunity here to extract the `Number(!applyIndentation)` expression and modify it so it depends on `firstItemOffset` instead of coercing a boolean. That way we could allow for simple configuration in the future in case we need to indent more than 1 step. WDYT?

Reply:
This is great idea, i've implemented changes. Do you think it would be wise to pass this as prop (itemStepOffset) instead of hardcoded configuration? That way if we want to achieve different offset we could just change prop value in place of implementation.

Reply:
I think it's a bit better to leave the boolean prop (and hardcoded config) for now in order to keep the component API minimal but either approach seems okay. You make the call 📞